misc/python/materialize/mzcompose.py

```diff
@@ -701,7 +703,58 @@ def __init__(
         command = f"--data-directory={data_directory} {options} --disable-telemetry --experimental --listen-addr 0.0.0.0:{port} --timestamp-frequency 100ms"
 
         config: PythonServiceConfig = (
-            {"image": image} if image else {"mzbuild": "materialized"}
+            {"image": image} if image else {"mzbuild": mzbuild}
+        )
+
+        if hostname:
+            config["hostname"] = hostname
+
+        # Depending on the docker-compose version, this may either work or be ignored with a warning
+        # Unfortunately no portable way of setting the memory limit is known
+        if memory:
+            config["deploy"] = {"resources": {"limits": {"memory": memory}}}
+
+        config.update(
+            {
+                "command": command,
+                "ports": [port],
+                "environment": environment,
+                "volumes": volumes,
+                "depends_on": depends_on,
+            }
+        )
+
+        super().__init__(name=name, config=config)
+
+
+class Dataflowd(PythonService):
```

Comment:
To avoid code duplication and to have symettry, consider one of the following:
- create Dataflowd and Coordd as subclasses to Materialized , so that most code, e.g. the ENV list, can be shared
- Do not have a separate class Dataflowd , but instead have everything in the `mzworkflows.py` be instances of  `Materialized`, just with a custom port, mzbuild name and command-line options each.

I can rearrange all that in a follow-up commit/PR if you prefer.

Reply:
I'd like to keep the `dataflowd` separate because it is an entirely different binary and will accept a entirely different set of options and environment parameters. I'll clean up the current definition so this will be clearer.